api/build.gradle

```diff
@@ -1,32 +1,74 @@
 description = "reactive-streams"
-def jdkFlow = false
+def isJdk9OrGreater = false
 try {
     Class.forName("java.util.concurrent.Flow")
-    jdkFlow = true
+    isJdk9OrGreater = true
 } catch (ClassNotFoundException cnfe) {
 
 }
 
 sourceSets {
     main {
         java {
-            if (jdkFlow)
-              srcDirs = ['src/main/java', 'src/main/java9']
-            else
-              srcDirs = ['src/main/java']
+            srcDirs = ['src/main/java']
         }
     }
+    if (isJdk9OrGreater) java9 {
+        java {
+            srcDirs = ['src/main/java', 'src/main/java9']
+        }
+    }
+}
+
+configurations {
+  mrjarArtifact
 }
```

Comment:
The MRJAR artifact is exported from the project, so that tests can depend on it. This is important to accurately test JPMS, otherwise Gradle will put these classes on the classpath downstream, which runs tests without modular encapsulation.